### DISCUSSION TITLE
All Quad Mesh

### DISCUSSION THREAD

**Discussion by angelmusonda** [FloSewn/TQMesh#21] [Q&A]
Hello everyone.

I am new to this library. But I want to find out if it can produce "all quad" meshes.

Thanks.

**Comment by FloSewn** [comment]
Hey,

currently you can produce all-quad meshes only by using "quad-refinements".
So basically, you triangulate your domain and then you divide each triangle in three quads.
Unfortunately this will probably not lead to the kind of all-quad mesh quality, that you would like to achieve. 

TQMesh also gives the opportunity to use a different meshing algorithm, which is called "Tri-to-Quad". Here we also start with a triangulated domain and then merge adjacent triangles to quads. However, with the current state of the algorithm, there might be some triangles left at the end of this stage and for that purpose you also need an additional "quad-refinement" to get rid of these last triangles. 
This approach is also shown in example 02.

Improving the "Tri-to-Quad" algorithm (also referred to as quad-morphing") is definitely on my to do list for coming improvements - as soon as I have some time left for it!

Anyway, thanks for your interest in this project! :-)

Regards, 
Flo

**Comment by angelmusonda** [comment]
Hello, thank you so much. I will try to work with that, I think it is sufficient for my needs.
